src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,838 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Device.Spi;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using global::Iot.Units;
+
+namespace Iot.Device.Ad7193
+{
+    /// <summary>
+    /// Represents the Analog Devices AD7193, the 4-channel, 4.8 kHz, ultralow noise, 24-bit sigma-delta ADC with PGA
+    /// </summary>
+    public class Ad7193 : IDisposable
+    {
+        #region Metadata for IDevice
+
+        /// <summary>
+        /// Name of the manufacturer of the device
+        /// </summary>
+        public const string Manufacturer = "Analog Devices";
+
+        /// <summary>
+        /// Name of the product
+        /// </summary>
+        public const string Product = "AD7193";
+
+        /// <summary>
+        /// Category of the device
+        /// </summary>
+        public const string ProductCategory = "ADC";
+
+        /// <summary>
+        /// Description of the device
+        /// </summary>
+        public const string ProductDescription = "4-Channel, 4.8 kHz, Ultralow Noise, 24-Bit Sigma-Delta ADC with PGA";
+
+        /// <summary>
+        /// The URI of the datasheet of the device
+        /// </summary>
+        public const string DataSheetURI = "https://www.analog.com/media/en/technical-documentation/data-sheets/AD7193.pdf";
+        #endregion
+
+        #region Metadata for ISpiDevice
+
+        /// <summary>
+        /// The list of SPI modes that are valid for this device
+        /// </summary>
+        public const SpiMode ValidSpiModes = SpiMode.Mode3;
```

Comment:
NIT: these two SPI related properties shouldn't be part of the public API, if you need them to do validation during execution just keep them as private or internal

Reply:
The SPI properties are related to the technical restriction we must apply if we use this particular device, so I think it is very good if we have this information on the class.
On the other hand, I agree that some check should be done relating to these settings in the constructor to check if the SpiDevice in the right mode/at the right frequency.